src/pybind/mgr/dashboard/frontend/src/app/shared/api/rbd-mirroring.service.spec.ts

```diff
@@ -1,34 +1,50 @@
-import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
-import { TestBed } from '@angular/core/testing';
+import { HttpClient } from '@angular/common/http';
+import { fakeAsync, TestBed, tick } from '@angular/core/testing';
+
+import { of as observableOf } from 'rxjs';
 
 import { configureTestBed } from '../../../testing/unit-test-helper';
 import { RbdMirroringService } from './rbd-mirroring.service';
 
 describe('RbdMirroringService', () => {
   let service: RbdMirroringService;
-  let httpTesting: HttpTestingController;
+
+  const summary = {
+    status: 0,
+    content_data: {
+      daemons: [],
+      pools: [],
+      image_error: [],
+      image_syncing: [],
+      image_ready: []
+    }
+  };
+
+  const httpClientSpy = {
+    get: () => observableOf(summary)
+  };
 
   configureTestBed({
-    providers: [RbdMirroringService],
-    imports: [HttpClientTestingModule]
+    providers: [RbdMirroringService, { provide: HttpClient, useValue: httpClientSpy }]
   });
 
   beforeEach(() => {
     service = TestBed.get(RbdMirroringService);
-    httpTesting = TestBed.get(HttpTestingController);
-  });
-
-  afterEach(() => {
-    httpTesting.verify();
   });
 
   it('should be created', () => {
     expect(service).toBeTruthy();
   });
 
-  it('should call get', () => {
-    service.get().subscribe();
-    const req = httpTesting.expectOne('api/rbdmirror');
-    expect(req.request.method).toBe('GET');
-  });
+  it('should call refresh', fakeAsync(() => {
+    let result = false;
+    service.refresh();
+    service.subscribe(() => {
+      result = true;
+    });
```

Comment:
I basically just copied the style from ```summary.service.spec.ts``` since that provided the example for me for how to do shared data service (to avoid each component individually make the same API request over-and-over again instead of sharing the data).

Reply:
I've looked into the file and you are right, it's not right there two. I will fix that. Thank you for the hint :)